client/src/connection_cache.rs

```diff
@@ -21,15 +28,131 @@ enum Connection {
     Quic(Arc<QuicTpuConnection>),
 }
 
+#[derive(Default)]
+struct MovingStatInner {
+    last: u64,
+    now: u64,
+}
+
+#[derive(Default)]
+pub struct MovingStat {
+    stats: Mutex<MovingStatInner>,
+}
+
+impl MovingStat {
+    pub fn update_stat(&self, new_value: u64) {
+        let mut stats = self.stats.lock().unwrap();
+        let diff = new_value.saturating_sub(stats.last);
+        // got an older update
+        if diff == 0 {
+            return;
+        }
+        stats.now = stats.now.saturating_add(diff);
+        stats.last = new_value;
+    }
+
+    pub fn load_and_reset(&self) -> u64 {
+        let mut stats = self.stats.lock().unwrap();
+        let old = stats.now;
+        stats.now = 0;
```

Comment:
Does stats.last need to be reset too?

Reply:
no, because it keeps track of where the counter is at. The counter data comes from the accumulation of the data from the quic counters which afaict dont' have a way to be reset, so in order to find the delta we keep track of the last value.